### PR TITLE
update balancer to accept json values for room metadata

### LINKED ISSUE #1339: balancer: fix message parsing error -- `invalid type: string "true"`
### Current Behavior

Started seeing this log message from the balancer

```
2024-02-16T14:03:12.261509Z ERROR ott_balancer::balancer: failed to dispatch monolith message e7768db2-da5a-4726-9c07-c98392daf866: invalid type: string "true", expected a boolean at line 1 column 116
```

### Expected Behavior

All messages from monoliths should be valid json, and the balancer should be able to parse them.

### Steps To Reproduce

1. `docker-compose -f docker/with-balancer.docker-compose.yml up -d --scale opentogethertube=10`
2. Watch logs with `docker-compose -f docker/with-balancer.docker-compose.yml logs -f --tail 1`
3. See log message

### Environment

- [ ] This happens on the official site, opentogethertube.com
- [ ] This happens using a self-hosted version.
- [ ] I'm using the docker image.

### Anything else?

I think this has something to do with the `M2BGossip` mesage

### PR DESCRIPTION
**Pull request by Victor-M-Giraldo**
Closes #1339

### CODE REVIEW

**Comment by Victor-M-Giraldo** [comment]
Didn't make name a `serde_json::Value` since I wasn't sure how I could convert it into a `RoomName` with the pointers and stuff.

**Review by dyc3 — APPROVED** [review]
Looks good.

**Comment by Victor-M-Giraldo** [comment]
Do you know why the lint is failing? I run yarn lint on my end and it goes through fine.

File: server/generated.ts [hunk 1]
```diff
@@ -51,14 +51,14 @@ export enum Visibility {
 /** Metadata about a room, according to the Monolith. */
 export interface RoomMetadata {
 	name: RoomName;
-	title: string;
-	description: string;
-	isTemporary: boolean;
+	title: unknown;
+	description: unknown;
+	isTemporary: unknown;
```
Comment (dyc3) [inline server/generated.ts:56]: Did you manually change these to unknown? Typeshare doesn't treat these as unknown by default, and it doesn't have an escape hatch to set the type manually. The codegen script has a hack to manually tweak the output of typeshare so it works a little better for us (that's why `currentSource` wasn't causing problems)

Comment (Victor-M-Giraldo) [reply server/generated.ts:56]: Yes, I ran the typeshare script and these didn't get updated (or well, they did but to `Value`)

Comment (dyc3) [reply server/generated.ts:56]: The CI job enforces that running the codegen script produces the exact same output as what is in `generated.ts`

File: crates/ott-balancer-protocol/src/monolith.rs [hunk 2]
```diff
@@ -206,16 +206,19 @@ where
...
+    #[typeshare(serialized_as = "unknown")]
     pub description: serde_json::Value,
-    #[serde(rename = "isTemporary")]
+    #[typeshare(serialized_as = "unknown")]
     pub is_temporary: serde_json::Value,
     pub visibility: Visibility,
-    #[serde(rename = "queueMode")]
+    #[typeshare(serialized_as = "unknown")]
     pub queue_mode: serde_json::Value,
-    #[serde(rename = "currentSource")]
```
Comment (dyc3) [inline crates/ott-balancer-protocol/src/monolith.rs:216]: these renames need to stay. you can put multiple attributes on a field

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: crates/ott-balancer-protocol/src/monolith.rs (hunk 2 above)
Comment (Victor-M-Giraldo) [reply crates/ott-balancer-protocol/src/monolith.rs:216]: ah, makes sense. thats why the names changed on the monolith end. will change that in a second.

**Review by dyc3 — APPROVED** [review]

**Review by dyc3 — APPROVED** [review]

**Pull request merged**